app/services/appropriate_bodies/process_batch/action.rb

```diff
@@ -0,0 +1,161 @@
+module AppropriateBodies
+  module ProcessBatch
+    # Management of induction periods in bulk via CSV upload
+    # Handles Pass / Fail / Release in two stages
+    class Action < Base
+      # @return [Array<?>] convert the valid submissions into permanent records
+      def do!
+        pending_induction_submission_batch.pending_induction_submissions.without_errors.map do |pending_induction_submission|
+          @pending_induction_submission = pending_induction_submission
+
+          do_action!
+        end
+      end
+
+      # @return [CSV::Table] validate each row and create a submission capturing the errors
+      def process!
+        pending_induction_submission_batch.rows.each do |row|
+          @row = row
+          @pending_induction_submission = sparse_pending_induction_submission
+
+          if (trs_error = fetch_trs_details!)
+            pending_induction_submission.update(error_message: trs_error)
+            next
+          end
+
+          if teacher.blank?
+            pending_induction_submission.update(error_message: "Teacher #{name} has not yet been claimed")
+            next
+          end
+
+          if ongoing_induction_period.blank?
+            pending_induction_submission.update(error_message: "Teacher #{name} does not have an ongoing induction")
+            next
+          end
+
+          if claimed_by_another_ab?
+            pending_induction_submission.update(error_message: "Teacher #{name} was claimed by another appropriate body")
```

Comment:
This line is repeated 4 times here, how do you feel about extracting it out into a private method?

Something like: 

```ruby
def reject_with_error(message)
  pending_induction_submission.update(error_message: message)
end
```